crates/reporting/src/error/type.rs

```diff
@@ -1167,14 +1167,22 @@ fn to_expr_report<'b>(
                             ]),
                             alloc.region(lines.convert_region(expr_region), severity),
                             match called_via {
-                                CalledVia::RecordBuilder => {
+                                CalledVia::OldRecordBuilder => {
                                     alloc.concat([
                                         alloc.tip(),
                                         alloc.reflow("Remove "),
                                         alloc.keyword("<-"),
                                         alloc.reflow(" to assign the field directly.")
                                     ])
                                 }
+                                CalledVia::NewRecordBuilder => {
+                                    alloc.concat([
+                                        alloc.note(""),
+                                        alloc.reflow("Record builders need a mapper function before the "),
+                                        alloc.keyword("<-"),
+                                        alloc.reflow(" to combine fields together with.")
+                                    ])
+                                }
```

Comment:
When does this happen? I didn't find it covered in the tests

Reply:
Yes, there isn't a test, I missed this one. Let me add a test case.